docs/transfers.rst

```diff
@@ -0,0 +1,296 @@
+   :scale: 100 %
+   :alt: Process of sending a transfer in IOTA.
+
+   Process of creating and sending a transfer to the Tangle.
+
+.. py:currentmodule:: iota
+
+1. Create Transactions
+~~~~~~~~~~~~~~~~~~~~~~
+The first step is to create the individual transaction objects. You have to
+specify ``address`` and ``value`` for each transaction. A negative ``value``
+means spending from ``address``. Furthermore, you can define a ``tag``, and for
+zero-value transactions, a ``message``. ``timestamp`` is usually auto-generated
+by the IOTA libraries.
+
+In PyOTA, use :py:class:`ProposedTransaction` to declare transactions.
+
+2. Create Bundle from Transactions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A bundle is a collection of transactions, treated as an atomic unit
+when sent to the network. A bundle makes a value (token) transfer possible by
+grouping together input an output transactions.
+
+A bundle always has to be balanced: the sum of ``value`` attributes of the
+transactions in the bundle should always be zero. Transactions in the bundle
+are indexed individually, and also contain information on how many other
+transactions there are in the bundle.
+
+Once complete, a bundle has to be finalized to generate the bundle hash based
+on the `bundle essence`_. The bundle hash is the unique identifier of the
+bundle.
+
+After finalization, input transactions in the bundle need to be signed to prove
+ownership of tokens being transferred.
+
+:py:class:`ProposedBundle` helps you in PyOTA to create bundles, add transactions,
+finalize the bundle and sign the inputs.
+
+3. Select two tips
+~~~~~~~~~~~~~~~~~~
+
+Tips are transactions that are yet to be confirmed by the network. We can
+obtain two tips by requesting them from a node. In PyOTA, :py:meth:`~Iota.get_transactions_to_approve`
+does the job: it returns a ``trunk`` and a ``branch`` :py:class:`TransactionHash`.
+
+Our bundle will validate these two transactions once in the Tangle.
+
+4. Do Proof-of-Work
+~~~~~~~~~~~~~~~~~~~
+
+The bundle has been finalized, inputs have been signed, we have two tips,
+now it's time to prepare the bundle to be attached to the Tangle. All
+transactions reference two other transactions in the Tangle, therefore we need
+to select these references for each transaction in our bundle.
+
+We also know that transactions `within the bundle are linked together`_ through
+their trunk references. So how do we construct the correct bundle structure
+and also reference two tips from the network?
+
+.. figure:: images/bundle-structure.png
+   :scale: 100 %
+   :alt: Bundle structure with four transactions.
+
+   Structure of a bundle with four transactions. Numbers in brackets denote
+   (``currentIndex``, ``lastIndex``) fields. Head of the bundle has index 3,
+   while tail has index 0.
+
+For all non-head transactions in the bundle, the trunk reference is the next
+transaction in the bundle, while the branch reference is the trunk transaction
+hash, one of the tips.
```

Comment:
```suggestion
transaction in the bundle, while the branch reference is one of the tips.
```

I thought that branch is the same for all transactions in a bundle (including the head tx)?

Reply:
It is not, according to https://docs.iota.org/docs/getting-started/0.1/transactions/bundles .

Reply:
Take a look at https://github.com/iotaledger/iri/blob/e1776fbad5d90df86a26402f9025e4b0b2ef7d3e/src/main/java/com/iota/iri/service/API.java#L1256-L1278
The result of this algo is the figure of the bundle structure on the docs site. And btw the documentation on top of the function in IRI is broken I think :smile_cat:

Reply:
Oh, that's interesting.  I didn't realise they were flipped 😬